src/Raven.Server/Documents/Handlers/SecretKeyHandler.cs

```diff
@@ -62,6 +62,9 @@ public unsafe Task PutKey()
             var name = GetStringQueryString("name");
             var overwrite = GetBoolValueQueryString("overwrite", required: false) ?? false;
 
+            if (Server.ServerCertificate == null)
```

Comment:
I'm not sure if this is correct. We might be running behind an HTTPS proxy here, so we aren't using HTTPS, but requests to us go over HTTPS.